### PR TITLE
feat: support MSW v2

### PR DESCRIPTION
**Pull request by ctuckz**
Fixes https://github.com/Tapico/tapico-msw-webarchive/issues/18. Happy to answer any questions!

Upgrades MSW to v2
- v2-style response resolver: https://mswjs.io/docs/migrations/1.x-to-2.x#response-resolver-arguments
- cookies are now set through response headers: https://mswjs.io/docs/migrations/1.x-to-2.x#ctxcookie
- new delay function: https://mswjs.io/docs/migrations/1.x-to-2.x#ctxdelay

Test updates
- upgraded jest to v29 to fix the issue from https://github.com/mswjs/msw/issues/1810
- updated json-transformer to be compatible with jest 29 (28, technically): https://jest-archive-august-2023.netlify.app/docs/28.x/upgrading-to-jest28#transformer
- added jest.polyfills.js to fix https://mswjs.io/docs/migrations/1.x-to-2.x#requestresponsetextencoder-is-not-defined-jest
- added testEnvironmentOptions in jest.config.js to fix https://mswjs.io/docs/migrations/1.x-to-2.x#cannot-find-module-mswnode-jsdom
- removed custom fetch implementation in utils/index.js to fix an issue with requests not being intercepted for some delay() values in `it('should handle request with content-encoding/compression response header from webarchive')`
  -  I'm still not quite sure why this test was failing. It would pass with a short delay (< 100ms) but consistently fail to intercept the call with a longer delay. When it failed to intercept the call, code would stop running in the resolver after `await delay` without throwing any errors. I found [this issue](https://github.com/mswjs/msw/issues/1930) which recommended removing node-fetch and that seems to work, but I think their problem was different than what I was seeing so I'm not 100% sure why
- Reset mocks after each test to prevent `expect(console.warn)` in `it('should remap origins if requested')` from catching log messages from `it('should warn when no definitions are given')`
- Changed `onUnhandledRequest` to `error` so that tests fail if a request isn't intercepted
- New tests for setting cookies, headers, and strict query string matching

### CODE REVIEW

**Review by tapico-weyert — APPROVED** [review]
Awesome work, thank you very much ❤️🤗

File: yarn.lock [hunk 1]
```diff
@@ -2,6 +2,14 @@
 # yarn lockfile v1
 
 
+"@ampproject/remapping@^2.2.0":
```
Comment (tapico-weyert) [inline yarn.lock:5]: Could it be that is part of some company artificatory instance?

Comment (ctuckz) [reply yarn.lock:5]: whoops, yes. Sorry about that!

Comment (tapico-weyert) [reply yarn.lock:10]: Thank you :)

**Comment by tapico-weyert** [comment]
Thank you! I think I have to fix the Github workflows in a separate PR

**Pull request merged**